src/CommonDBTM.php

```diff
@@ -5855,7 +5855,7 @@ private function assetBusinessRules($condition)
             }
 
             // Set the condition (add or update)
-            $output = $ruleasset->processAllRules($input, [], [], [
```

Comment:
Please note this is probably a big change, but I struggle to estimate properly.
It aligns the behavior with RuleTicket as you can see here https://github.com/glpi-project/glpi/blob/aad5067a705333d5ebbf06d0dd08910a756cf79a/src/CommonITILObject.php#L11250-L11255

Without passing the input to the &output parameter, 'append' group action behaves like an 'assign', no previous groups were passed.